addons/etcd/scripts/switchover-with-candidate.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+set -ex
+
+leaderEndpoint=${KB_LEADER_POD_FQDN}:2379
+candidateEndpoint=${KB_SWITCHOVER_CANDIDATE_FQDN}:2379
+
+# see common.sh, this function may change leaderEndpoint
+updateLeaderIfNeeded 3
+
```

Comment:
maybe after updateLeaderIfNeeded, leaderEndpoint already equals to candidateEndpoint.

should add a check again

Reply:
In fact, I have check it inside the updateLeaderIfNeeded function, I know it's not really elegantly implementation, but if I add the input and output args to this function, it will look like:
`leaderEndpoint=$(updateLeaderIfNeeded peerEndpoints leaderEndpoint 3)`
and I can't simple output error message, because shell command will treat each as output, so I check it and return inside this function

Reply:
extracted it outside